services/sources/publisher_backlist_service.py

```diff
@@ -0,0 +1,110 @@
+from datetime import datetime, timedelta, timezone
+import os
+import requests
+import json
+import urllib.parse
+from typing import Optional, Dict
+
+from logger import create_log
+from mappings.UofM import UofMMapping
+from .source_service import SourceService
+
+
+logger = create_log(__name__)
+
+BASE_URL = "https://api.airtable.com/v0/appBoLf4lMofecGPU/Publisher%20Backlists%20%26%20Collections%20%F0%9F%93%96?view=All%20Lists"
+
+class PublisherBacklistService(SourceService):
+    def __init__(self):
+        self.airtable_auth_token = os.environ.get('AIRTABLE_KEY', None)
+
+    def get_records(
+        self,
+        full_import: bool=False, 
+        start_timestamp: datetime=None,
+        offset: Optional[int]=None,
+        limit: Optional[int]=None
+    ) -> list[UofMMapping]:
+        
+        array_json_records = self.get_records_json(full_import, start_timestamp, offset, limit)
+
+
+        for json_dict in array_json_records:
+            for records_value in json_dict['records']:
```

Comment:
I recommend that get_records_array does this logic to return the records. that way this function is only getting back a list of json records and has to simply map it.

Reply:
I think I need to clarify what array_json_records consists of. It seems from what you're proposing that array_json_records looks like this:

`[{"records": [values from all records from airtable API]}]`

but in actuality it looks like this:

`[{"records": [value from first 100 records from airtable]}, {"records":[value from next 100 records from airtable]}]`

That's why I can't just return the values from the records key one time since there are several records due to their being multiple dictionaries in the array.